include/envoy/local_info/local_info.h

```diff
@@ -25,17 +25,17 @@ class LocalInfo {
   /**
    * Human readable zone name. E.g., "us-east-1a".
    */
-  virtual const std::string& zoneName() const PURE;
+  virtual const std::string zoneName() const PURE;
```

Comment:
This whole situation continues to make me sad, but I don't think there is anything I can do about it without hindering progress. Can you put a TODO in here with a link to this PR so that we can revert this when you sort out your internal string discrepancies?